src/main/java/org/apache/pirk/encryption/Paillier.java

```diff
@@ -340,4 +342,41 @@ private String parametersToString()
     return "p = " + p.intValue() + " q = " + q.intValue() + " N = " + N.intValue() + " NSquared = " + NSquared.intValue() + " lambdaN = " + lambdaN.intValue()
         + " bitLength = " + bitLength;
   }
+
+  @Override public boolean equals(Object o)
+  {
+    if (this == o)
+      return true;
+    if (o == null || getClass() != o.getClass())
+      return false;
+
+    Paillier paillier = (Paillier) o;
+
+    if (bitLength != paillier.bitLength)
+      return false;
+    if (!p.equals(paillier.p))
+      return false;
+    if (!q.equals(paillier.q))
+      return false;
+    if (!N.equals(paillier.N))
+      return false;
+    if (!NSquared.equals(paillier.NSquared))
+      return false;
+    if (!lambdaN.equals(paillier.lambdaN))
+      return false;
```

Comment:
I can, but it just leaves a blob of ORd-together stuff. I admit to knowing little about the java compiler; wouldn't it just optimize these into such a beast anyway? If so, then I feel like it's more aesthetically pleasing to just leave them separated. If not, well, screw that, and I'll change it; or if the community feels my subjective aesthetics aren't aesthetically pleasing.